applications/serial_lte_modem/src/main.c

```diff
@@ -55,16 +55,6 @@ static void indicate_wk(struct k_work *work);
 
 BUILD_ASSERT(CONFIG_SLM_WAKEUP_PIN >= 0, "Wake up pin not configured");
 
-NRF_MODEM_LIB_ON_INIT(serial_lte_modem_init_hook, on_modem_lib_init, NULL);
-
-/* Initialized to value different than success (0) */
-static int modem_lib_init_result = -1;
-
-static void on_modem_lib_init(int ret, void *ctx)
-{
-	modem_lib_init_result = ret;
-}
-
```

Comment:
This code is for modem fault hander in application, cannot be removed

Reply:
I'm not following. The `modem_lib_init_result` is received when the modem library is initialized. As we now call `nrf_modem_lib_init()` from `main`, we can get the initialization result directly by the return value, and pass it to `handle_nrf_modem_lib_init_ret(int ret)` directly.

Reply:
Sorry, misunderstood. 
Outside `#if defined(CONFIG_NRF_MODEM_LIB_ON_FAULT_APPLICATION_SPECIFIC)` scope so just fine.
The re-initiate is not in main() but in on_modem_failure_reinit().